File: src/main/java/com/github/motui/meican/order/OrderToolWindowFactory.java
package com.github.motui.meican.order;

import com.intellij.openapi.project.DumbAware;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.wm.ToolWindow;
import com.intellij.openapi.wm.ToolWindowFactory;
import com.intellij.ui.content.Content;
import com.intellij.ui.content.ContentFactory;
import org.jetbrains.annotations.NotNull;

/**
 * @author motui
 * @date 2021-01-16
 */
public class OrderToolWindowFactory implements ToolWindowFactory, DumbAware {
  @Override
  public void createToolWindowContent(@NotNull Project project, @NotNull ToolWindow toolWindow) {
    OrderWindow orderWindow = new OrderWindow();
    ContentFactory contentFactory = ContentFactory.SERVICE.getInstance();
    Content content = contentFactory.createContent(orderWindow.getContent(), "", false);
    toolWindow.getContentManager().addContent(content);
  }
}

File: src/main/kotlin/cn/motui/meican/ui/order/OrderToolWindowFactory.kt
package cn.motui.meican.ui.order

import com.intellij.openapi.project.DumbAware
import com.intellij.openapi.project.Project
import com.intellij.openapi.wm.ToolWindow
import com.intellij.openapi.wm.ToolWindowFactory

/**
 * 工具窗口工厂
 */
class OrderToolWindowFactory : ToolWindowFactory, DumbAware {
    override fun createToolWindowContent(project: Project, toolWindow: ToolWindow) {
        OrderView.instance.setup(project, toolWindow)
    }
}
